// nuxi/dist/dev/index.mjs
import "../consola.DXBYu-KD-BxkQ5_xB.mjs";
import "../utils-D5jIbZ8_.mjs";
import "../dist-CGV6T3ee.mjs";
import "../logger-TGEXLs5w.mjs";
import { t as defu } from "../defu-DcpjBOZc.mjs";
import { t as overrideEnv } from "../env-Dgf3oimn.mjs";
import "../kit-BqwTIe0r.mjs";
import { t as NuxtDevServer } from "../utils--eSm-kwO.mjs";
import "../satisfies--kW1HehR.mjs";
import "../versions-CItnJg_E.mjs";
import "../banner-DR0HFUbH.mjs";
import "../fs-CN5CWQ-H.mjs";
import "../nuxt-ISq-9kSI.mjs";
import process from "node:process";

//#region src/dev/index.ts
const start = Date.now();
var IPC = class {
	enabled = !!process.send && !process.title?.includes("vitest") && process.env.__NUXT__FORK;
	constructor() {
		if (this.enabled) process.once("unhandledRejection", (reason) => {
			this.send({
				type: "nuxt:internal:dev:rejection",
				message: reason instanceof Error ? reason.toString() : "Unhandled Rejection"
			});
			process.exit();
		});
		process.on("message", (message) => {
			if (message.type === "nuxt:internal:dev:context") initialize(message.context, { listenOverrides: message.listenOverrides });
		});
		this.send({ type: "nuxt:internal:dev:fork-ready" });
	}
	send(message) {
		if (this.enabled) process.send?.(message);
	}
};
const ipc = new IPC();
async function initialize(devContext, ctx = {}) {
	overrideEnv("development");
	const devServer = new NuxtDevServer({
		cwd: devContext.cwd,
		overrides: defu(ctx.data?.overrides, { extends: devContext.args.extends }),
		logLevel: devContext.args.logLevel,
		clear: devContext.args.clear,
		dotenv: {
			cwd: devContext.cwd,
			fileName: devContext.args.dotenv
		},
		envName: devContext.args.envName,
		showBanner: ctx.showBanner !== false && !ipc.enabled,
		listenOverrides: ctx.listenOverrides
	});
	let address;
	if (ipc.enabled) {
		devServer.on("loading:error", (_error) => {
			ipc.send({
				type: "nuxt:internal:dev:loading:error",
				error: {
					message: _error.message,
					stack: _error.stack,
					name: _error.name,
					code: "code" in _error ? _error.code : void 0
				}
			});
		});
		devServer.on("loading", (message) => {
			ipc.send({
				type: "nuxt:internal:dev:loading",
				message
			});
		});
		devServer.on("restart", () => {
			ipc.send({ type: "nuxt:internal:dev:restart" });
		});
		devServer.on("ready", (payload) => {
			ipc.send({
				type: "nuxt:internal:dev:ready",
				address: payload
			});
		});
	} else devServer.on("ready", (payload) => {
		address = payload;
	});
	await devServer.init();
	if (process.env.DEBUG) console.debug(`Dev server (internal) initialized in ${Date.now() - start}ms`);
	return {
		listener: devServer.listener,
		close: async () => {
			devServer.closeWatchers();
			await Promise.all([devServer.listener.close(), devServer.close()]);
		},
		onReady: (callback) => {
			if (address) callback(address);
			else devServer.once("ready", (payload) => callback(payload));
		},
		onRestart: (callback) => {
			let restarted = false;
			function restart() {
				if (!restarted) {
					restarted = true;
					callback(devServer);
				}
			}
			devServer.once("restart", restart);
			process.once("uncaughtException", restart);
			process.once("unhandledRejection", restart);
		}
	};
}

//#endregion
export { initialize };
