// nuxi/dist/generate-CO4rPHRv.mjs
import { t as defineCommand } from "./dist-sJJifug_.mjs";
import "./consola.DXBYu-KD-BxkQ5_xB.mjs";
import "./utils-D5jIbZ8_.mjs";
import { a as legacyRootDirArgs, i as extendsArgs, n as dotEnvArgs, o as logLevelArgs, r as envNameArgs, t as cwdArgs } from "./_shared-CoPxtaYg.mjs";
import "./logger-TGEXLs5w.mjs";
import "./env-Dgf3oimn.mjs";
import "./kit-BqwTIe0r.mjs";
import "./satisfies--kW1HehR.mjs";
import "./versions-CItnJg_E.mjs";
import "./banner-DR0HFUbH.mjs";
import "./fs-CN5CWQ-H.mjs";
import build_default from "./build-BJVzIIXr.mjs";

//#region src/commands/generate.ts
var generate_default = defineCommand({
	meta: {
		name: "generate",
		description: "Build Nuxt and prerender all routes"
	},
	args: {
		...cwdArgs,
		...logLevelArgs,
		preset: {
			type: "string",
			description: "Nitro server preset"
		},
		...dotEnvArgs,
		...envNameArgs,
		...extendsArgs,
		...legacyRootDirArgs
	},
	async run(ctx) {
		ctx.args.prerender = true;
		await build_default.run(ctx);
	}
});

//#endregion
export { generate_default as default };
